demos/multi_material/compositional_buoyancy/compositional_buoyancy.py

```diff
@@ -62,24 +64,24 @@
 # solutions, as in our previous tutorials.
 
 # +
-nx, ny = 40, 40  # Number of cells in x and y directions
-lx, ly = 0.9142, 1  # Domain dimensions in x and y directions
+mesh_elements = (64, 64)  # Number of cells in x and y directions
+domain_dims = (0.9142, 1.0)  # Domain dimensions in x and y directions
 # Rectangle mesh generated via Firedrake
-mesh = RectangleMesh(nx, ny, lx, ly, quadrilateral=True)
-mesh.cartesian = True
-boundary = get_boundary_ids(mesh)
+mesh = RectangleMesh(*mesh_elements, *domain_dims, quadrilateral=True)
+mesh.cartesian = True  # Tag the mesh as Cartesian to inform other G-ADOPT objects
+boundary = get_boundary_ids(mesh)  # Object holding references to mesh boundary IDs
 
-V = VectorFunctionSpace(mesh, "CG", 2)  # Velocity function space (vector)
-W = FunctionSpace(mesh, "CG", 1)  # Pressure function space (scalar)
+V = VectorFunctionSpace(mesh, "Q", 2)  # Velocity function space (vector)
+W = FunctionSpace(mesh, "Q", 1)  # Pressure function space (scalar)
 Z = MixedFunctionSpace([V, W])  # Stokes function space (mixed)
-Q = FunctionSpace(mesh, "CG", 2)  # Temperature function space (scalar)
+Q = FunctionSpace(mesh, "DQ", 2)  # Temperature function space (scalar)
```

Comment:
"Temperature" function space? Does this example have a temperature?

Reply:
And we have gone around full circle! Haha

Reply:
No, it does not. But `StokesSolver` still requires a `Function` for temperature as of today...

Reply:
I think of it this way: `StokesSolver` requires a temperature, so I need to define a temperature `Function`, so I need to define a temperature `FunctionSpace`. There is no reason for temperature and level-set function spaces to be the same.